### PR TITLE
Add check_existing_db_1 test

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add check_existing_db_1 test with ledger nano secret manager

## Links to any relevant issues

https://github.com/iotaledger/iota-sdk/issues/429

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Running the test

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: sdk/tests/wallet/wallet_storage.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ use iota_sdk::{
 
 use crate::wallet::common::{setup, tear_down};
 
+// Db created with commit 8dd389ddeed0d95bb493c38f376b41a6a9127148
```
Comment (thibault-martinez) [inline sdk/tests/wallet/wallet_storage.rs]:
```suggestion
// Db created with wallet.rs commit 8dd389ddeed0d95bb493c38f376b41a6a9127148
```

File: sdk/tests/wallet/wallet_storage.rs [hunk 2]
```diff
@@ -85,6 +86,55 @@ async fn check_existing_db() -> Result<()> {
     tear_down(storage_path)
 }
 
+// Db created with commit 2dd9974c1bc05c2b0b7d6f0ee100deb2da60d071
```
Comment (thibault-martinez) [inline sdk/tests/wallet/wallet_storage.rs]:
```suggestion
// Db created with wallet.rs commit 2dd9974c1bc05c2b0b7d6f0ee100deb2da60d071
```

**Review by thibault-martinez — DISMISSED** [review]

**Review by qrayven — DISMISSED** [review]

**Review by kwek20 — DISMISSED** [review]

File: sdk/tests/wallet/wallet_storage.rs [hunk 3]
```diff
@@ -85,6 +86,55 @@ async fn check_existing_db() -> Result<()> {
...
+    setup(storage_path)?;
+    // Copy db so the original doesn't get modified
+    copy_folder("./tests/wallet/fixtures/check_existing_1_db_test", storage_path).unwrap();
+
+    let wallet = Wallet::builder().with_storage_path(storage_path).finish().await?;
+
+    matches!(
+        *wallet.get_secret_manager().read().await,
+        iota_sdk::client::secret::SecretManager::LedgerNano(_)
+    );
```
Comment (Alex6323) [inline sdk/tests/wallet/wallet_storage.rs]: this does not make the test fail if it doesn't match

Comment (thibault-martinez) [reply sdk/tests/wallet/wallet_storage.rs]: Good catch, should probably have an assert on it

Comment (thibault-martinez) [reply sdk/tests/wallet/wallet_storage.rs]: I'll add it quickly, not sure if Thoralf is around

Comment (Alex6323) [reply sdk/tests/wallet/wallet_storage.rs]: wow 3 approvals, I must be missing something :trollface:

Comment (Alex6323) [reply sdk/tests/wallet/wallet_storage.rs]: haha, okay

Comment (thibault-martinez) [reply sdk/tests/wallet/wallet_storage.rs]: Pinging people works

Comment (thibault-martinez) [reply sdk/tests/wallet/wallet_storage.rs]: Done

**Review by Alex6323 — APPROVED** [review]
first :stuck_out_tongue_winking_eye:

**Review by kwek20 — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

File: sdk/tests/wallet/wallet_storage.rs (hunk 3 above)
Comment (Thoralf-M) [reply sdk/tests/wallet/wallet_storage.rs]: Thanks

Comment (thibault-martinez) [reply sdk/tests/wallet/wallet_storage.rs]: I'll allow myself to resolve this

**Pull request merged**